backend/services/question/src/controller.ts

```diff
@@ -90,7 +94,81 @@ export const getQuestions = async (req: Request, res: Response) => {
       .skip(offset * size) // skip the first offset * size elements
       .limit(size); // take the first (size) elements
 
-    res.status(200).json({ content: questions, total: total });
+    if (req.query.onlyUnattempted && req.query.user) {
+      const onlyUnattempted = req.query.onlyUnattempted === "true";
+      const resp = await fetch(
+        `${process.env.USER_API}/users/${req.query.user}/question-attempt`
+      );
+      const attemptedQuestionIds = new Set<Number>(await resp.json());
+
+      const modifiedQuestions = [];
+
+      for (const question of questions) {
+        const questionObject = question.toObject();
+        const wasAttempted = attemptedQuestionIds.has(questionObject.id);
+
+        if ((onlyUnattempted && !wasAttempted) || !onlyUnattempted) {
+          modifiedQuestions.push({
+            ...questionObject,
+            wasAttempted: wasAttempted,
+          });
+        }
+      }
+      res
+        .status(200)
+        .json({ content: modifiedQuestions, total: modifiedQuestions.length });
```

Comment:
Pagination must be accounted for with unattempted questions - total should be the total number of unattempted questions in the entire database for a user (consistent with how it is if unattempteds are not requested). also cannot simply filter an original page from the DB, and send that, as the number of entries in that page will now no longer be correct as per the number of items per page. There could be unattempteds in later pages that could be added to this page to meet the page number specified

Reply:
a way to think about it is like this: imagine the unattempted questions for user1 as its own database, and your perform retrieval of pages from there. Pagination values and number of records returned should be consistent with this logic